litestar/datastructures/internal.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Any
+
+from litestar.constants import CONNECTION_STATE_KEY
+from litestar.types import Empty, EmptyType
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from litestar.datastructures import URL, Accept, Headers
+    from litestar.types.asgi_types import Scope
+
+
+@dataclass
+class ConnectionState:
```

Comment:
Why the change in naming here from "scope state" to "connection state"? To me "scope state" is a bit more intuitive because it describes more accurately what this is: It takes a `Scope` and stores state in it

Reply:
I'm happy to name it anything

Reply:
That would be a terrible name :upside_down_face: 

Well, I'm +1 on `ScopeState`, but maybe other members have opinions on this as well?
@guacs perhaps? :)

Reply:
I'm happy with either name as well. Though I disagree with you, @provinzkraut, that `anything` is a terrible name. It would be a fantastic name :P

Reply:
> Though I disagree with you, @provinzkraut, that `anything` is a terrible name. It would be a fantastic name :P

Reminds me of the `Thing` class in the old Reddit codebase.

Let's leave the naming decision up to @peterschutt then if no one has a strong opinion here :)

Reply:
I'll go with `ScopeState` as that is the existing terminology and otherwise it feels a bit like change for the sake of change.